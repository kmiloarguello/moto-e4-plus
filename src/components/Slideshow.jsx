import React, { Component } from "react";
import styles from "../css/index.css";

export default class Slideshow extends Component {
  render() {
    return (
      <div className={styles.slide + " " + styles.fade}>
        <div className={styles.column}>
          <div className={styles.A}>
            <h2 style={{ color: this.props.color }} >{this.props.title}</h2>
          </div>
          <div className={styles.B}>
            <p style={{ color: this.props.color }} >{this.props.description}</p>
          </div>
          <div className={styles.C}>
            <img src={this.props.img} alt={this.props.title} />
          </div>
        </div>
      </div>
    );
  }
}
