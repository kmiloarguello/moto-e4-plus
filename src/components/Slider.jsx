import React, { Component } from "react";
import styles from "../css/index.css";
import Slideshow from "./Slideshow.jsx";

import s1 from "../arquivos/img/section4/s1.png";
import s2 from "../arquivos/img/section4/s2.png";
import s3 from "../arquivos/img/section4/s3.png";
import s4 from "../arquivos/img/section4/s4.png";
import s5 from "../arquivos/img/section4/s5.png";

import bg_s1 from "../arquivos/img/section4/bg-s1.png";
import bg_s2 from "../arquivos/img/section4/bg-s2.png";
import bg_s3 from "../arquivos/img/section4/bg-s3.png";
import bg_s4 from "../arquivos/img/section4/bg-s4.png";
import bg_s5 from "../arquivos/img/section4/bg-s5.png";

export default class Slider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      slideIndex: 1,
      isWhite: false,
      color: "#333f48"
    };
  }
  componentDidMount() {
    this.showSlides(this.state.slideIndex);
  }
  currentSlide(n) {
    this.showSlides((this.state.slideIndex = n));
  }
  showSlides(n) {
    let i;
    let slides = document.getElementsByClassName(styles.slide);
    let dots = document.getElementsByClassName(styles.dot);
    let bg = document.getElementsByClassName(styles.section4)[0];

    if (n > slides.length) {
      this.setState({
        slideIndex: 1
      });
    }
    if (n < 1) {
      this.setstate({
        slideIndex: slides.length
      });
    }
    for (i = 0; i < slides.length; i++) {
      slides[i].style.display = "none";
    }
    for (i = 0; i < dots.length; i++) {
      dots[i].className = dots[i].className.replace(" active", "");
    }
    slides[this.state.slideIndex - 1].style.display = "block";
    dots[this.state.slideIndex - 1].className += " active";

    /**
     * To do: Refactor of this
     */
    if (this.state.slideIndex == 1) {
      bg.style.backgroundImage = "url(" + bg_s1 + ")";
      this.setState({
        color: "#333f48"
      });
    } else if (this.state.slideIndex == 2) {
      bg.style.backgroundImage = "url(" + bg_s2 + ")";
      this.setState({
        color: "white"
      });
    } else if (this.state.slideIndex == 3) {
      bg.style.backgroundImage = "url(" + bg_s3 + ")";
      this.setState({
        color: "white"
      });
    } else if (this.state.slideIndex == 4) {
      bg.style.backgroundImage = "url(" + bg_s4 + ")";
      this.setState({
        color: "white"
      });
    } else {
      bg.style.backgroundImage = "url(" + bg_s5 + ")";
      this.setState({
        color: "#333f48"
      });
    }
  }
  render() {
    return (
      <section className={styles.section4}>
        <div className={styles.row}>
          <Slideshow
            title="Grande y brillante"
            description="Cuando veas la vibrante pantalla de 5.5'' y su increíble diseño metálico, no querrás dejarla. Elige entre los sofisticados acabados en gris metálico, y dorado. "
            img={s1}

            color={this.state.color}
          />
          <Slideshow
            title="Sonrisas por todas partes"
            description="Nunca te pierdas un momento. La cámara de enfoque automático de 13 MP toma excelentes fotos, incluso con poca luz. Además, toma selfies dignas de compartir gracias a una cámara de 5MP con su flash para selfies. "
            img={s2}

            color={this.state.color}
          />
          <Slideshow
            title="Espera lo mejor"
            description={[
              "Funciona con Android",
              <sup>TM</sup>,
              " 7.1, una versión actualizada del sistema operativo más famosos del mundo. Ahora gastar menos no es sinónimo de conformarse con menos."
            ]}
            img={s3}
            color={this.state.color}
          />
          <Slideshow
            title="Super potente"
            description={[
              "Gracias a la batería de 5000 mAh, puedes usarlo días sin preocuparte por la duración de la batería.1 Y cuando sea hora de recargar, no bajes el ritmo. El ",
              <strong>Modo E Plus</strong>,
              "funciona con un cargador rápido de 10W.",
              <br />,
              "*El cargador rápido se vende por separado."
            ]}
            img={s4}
            color={this.state.color}
          />
          <Slideshow
            title="Toca y continúa"
            description="Ya no te preocupes si olvidas la contraseña sólo usa tu huella digital y desbloquea el smartphone para que puedas acceder a lo que más te gusta con mayor rapidez."
            img={s5}
            color={this.state.color}
          />
          <div id={styles.dot_container}>
            <span className={styles.dot} onClick={() => this.currentSlide(1)} />
            <span className={styles.dot} onClick={() => this.currentSlide(2)} />
            <span className={styles.dot} onClick={() => this.currentSlide(3)} />
            <span className={styles.dot} onClick={() => this.currentSlide(4)} />
            <span className={styles.dot} onClick={() => this.currentSlide(5)} />
          </div>
        </div>
      </section>
    );
  }
}
