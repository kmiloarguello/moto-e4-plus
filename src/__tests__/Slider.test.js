import React from "react";
import Slider from "../components/Slider.jsx";
import Slideshow from "../components/Slideshow.jsx";
import { render, fireEvent } from "react-testing-library";
import styles from "../css/index.css";
import myImage from "../arquivos/img/section1/front.png";

describe("Slider section", () => {
  test("Slider section render correctly", () => {
    const { debug } = render(<Slider />);
  });
  test("Slideshow props updated correctly", () => {
    const { rerender } = render(
      <Slideshow
        title="titlee"
        description="description"
        img=""
        color={"red"}
      />
    );
    rerender(<Slideshow color={"green"} />);
  });
  test("Slideshow unmount correctly", () => {
    const { container, unmount } = render(<Slideshow />);
    unmount();
  });
  test("Slideshow has title and description", () => {
    const { getByTestId } = render(
      <Slideshow>
        <h2 data-testid="title-slideshow">Titulo</h2>
        <p data-testid="description-slideshow">Descrición</p>
      </Slideshow>
    );
    const title = getByTestId("title-slideshow");
    const description = getByTestId("description-slideshow");
  });
  test("Slideshow has alt image", () => {
    const { getByAltText } = render(
      <Slideshow title="Titulo" description="Descripción" img="" color="red" />
    );
    const altSlideshow = getByAltText("Titulo");
  });
  test("Span button trigger an event", () => {
    const spy = jest.fn();
    const { getByTestId } = render(<Slider />);

    fireEvent.click(getByTestId("dot-a"), spy);
  });
  test("Slider change of background", () => {
    const { getByTestId } = render(<Slider />);
    const bg = document.getElementsByClassName(styles.section4)[0];
    bg.style.backgroundImage = "url(" + myImage + ")";
    fireEvent.change(bg);
    expect(bg.style.backgroundImage).toBe("url(" + myImage + ")")
  });
});
