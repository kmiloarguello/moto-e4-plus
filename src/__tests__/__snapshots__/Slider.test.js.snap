// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider section Slider section snapshot 1`] = `
<section>
  <div>
    <div
      class="undefined "
      style="display: none;"
    >
      <div>
        <div>
          <h2
            data-testid="title-slideshow"
            style="color: rgb(51, 63, 72);"
          >
            Grande y brillante
          </h2>
        </div>
        <div>
          <p
            data-testid="description-slideshow"
            style="color: rgb(51, 63, 72);"
          >
            Cuando veas la vibrante pantalla de 5.5'' y su increíble diseño metálico, no querrás dejarla. Elige entre los sofisticados acabados en gris metálico, y dorado. 
          </p>
        </div>
        <div>
          <img
            alt="Grande y brillante"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
    <div
      class="undefined "
      style="display: none;"
    >
      <div>
        <div>
          <h2
            data-testid="title-slideshow"
            style="color: rgb(51, 63, 72);"
          >
            Sonrisas por todas partes
          </h2>
        </div>
        <div>
          <p
            data-testid="description-slideshow"
            style="color: rgb(51, 63, 72);"
          >
            Nunca te pierdas un momento. La cámara de enfoque automático de 13 MP toma excelentes fotos, incluso con poca luz. Además, toma selfies dignas de compartir gracias a una cámara de 5MP con su flash para selfies. 
          </p>
        </div>
        <div>
          <img
            alt="Sonrisas por todas partes"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
    <div
      class="undefined "
      style="display: none;"
    >
      <div>
        <div>
          <h2
            data-testid="title-slideshow"
            style="color: rgb(51, 63, 72);"
          >
            Espera lo mejor
          </h2>
        </div>
        <div>
          <p
            data-testid="description-slideshow"
            style="color: rgb(51, 63, 72);"
          >
            Funciona con Android 7.1, una versión actualizada del sistema operativo más famosos del mundo. Ahora gastar menos no es sinónimo de conformarse con menos.
          </p>
        </div>
        <div>
          <img
            alt="Espera lo mejor"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
    <div
      class="undefined "
      style="display: none;"
    >
      <div>
        <div>
          <h2
            data-testid="title-slideshow"
            style="color: rgb(51, 63, 72);"
          >
            Super potente
          </h2>
        </div>
        <div>
          <p
            data-testid="description-slideshow"
            style="color: rgb(51, 63, 72);"
          >
            Gracias a la batería de 5000 mAh, puedes usarlo días sin preocuparte por la duración de la batería.1 Y cuando sea hora de recargar, no bajes el ritmo. El Modo E Plus funciona con un cargador rápido de 10W. *El cargador rápido se vende por separado.
          </p>
        </div>
        <div>
          <img
            alt="Super potente"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
    <div
      class="undefined "
      style="display: none;"
    >
      <div>
        <div>
          <h2
            data-testid="title-slideshow"
            style="color: rgb(51, 63, 72);"
          >
            Toca y continúa
          </h2>
        </div>
        <div>
          <p
            data-testid="description-slideshow"
            style="color: rgb(51, 63, 72);"
          >
            Ya no te preocupes si olvidas la contraseña sólo usa tu huella digital y desbloquea el smartphone para que puedas acceder a lo que más te gusta con mayor rapidez.
          </p>
        </div>
        <div>
          <img
            alt="Toca y continúa"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
    <div
      data-testid="dots"
    >
      <span
        data-testid="dot-a"
      />
      <span
        data-testid="dot-b"
      />
      <span
        data-testid="dot-c"
      />
      <span
        data-testid="dot-d"
      />
      <span
        data-testid="dot-e"
      />
    </div>
  </div>
</section>
`;
